src/State.sol

```diff
@@ -52,6 +53,7 @@ abstract contract State is OwnableUpgradeable, ReentrancyGuardUpgradeable {
     uint256 public maxTotalAssetsInUnderlying;
 
     mapping(bytes4 => bool) public _isFunctionDisabled;
+    IWhitelistRegistry public whitelistRegistry;
```

Comment:
You add only whitelist registry. There is no bool var whitelistEnebled. And there is no bool inside whitelistRegistry

there is a strange thing - address(whitelistRegistry) == address(0) but it's very implicit, not explicit